src/utils/garminUtils.ts

```diff
@@ -27,6 +29,37 @@ function convertGarminRecord(garminRecord: GarminActivityRecord): Record {
   return { time, coord, altitude };
 }
 
+function isValidRecord(
+  record: Record,
+  index: number,
+  records: Record[]
+): boolean {
+  return (
+    true && isDistancePossible(record, index, records)
+    // add new checks here
+  );
+}
+
+export function isDistancePossible(
+  record: Record,
+  index: number,
+  records: Record[]
+): boolean {
+  // impossible means that this point is too far from the previous point
+  // to be possible for anybody (e.g. i = 0 in America, i = 1 in Spain, i >= 2 in America)
+  // Special case: first point is always possible
+  if (index === 0) {
+    return true;
+  }
+
+  // Assume the first coordinate is good
+  const goodCoordinate = records[0].coord;
+  const currentCoordinates = record.coord;
+  const distance = lawOfCosinesDistance(goodCoordinate, currentCoordinates);
+
+  return distance <= IMPOSSIBLE_INCREMENTAL_DISTANCE_METERS;
+}
```

Comment:
Doesn't this function only check if the distance between the first point and the `index`th point is below the threshold? There's no comparison to the previous point as far as I can tell; it always compares to the first point. That probably also works as a filtering method, unless the activity is a month-long spiritual journey

Reply:
Yeah good catch, I had changed the implementation but forgot the comments. Updated